adb_graphics/datahandler/gribdata.py

```diff
@@ -614,114 +471,103 @@ def fire_weather_index(self, values, **kwargs):
-            grid_info['projection'] = 'lcc'
-            grid_info['lat_0'] = 39.0
-        elif self.grid_suffix == 'GST0':
-            attrs = ['Lov']
-            grid_info['projection'] = 'stere'
-            grid_info['lat_0'] = 90
-        elif self.grid_suffix == 'GLL0':
-            attrs = []
-            grid_info['projection'] = 'cyl'
         else:
-            attrs = []
-            grid_info['projection'] = 'rotpole'
-
-            # CenterLon in RAP and Longitude_of_southern_pole in RRFS
-            lon_0 = lat.attrs.get('CenterLon', lat.attrs.get('Longitude_of_southern_pole'))
-            grid_info['lon_0'] = lon_0[0] - 360
-
-            # CenterLat in RAP and Latitude_of_southern_pole in RRFS
-            center_lat = lat.attrs.get('CenterLat', lat.attrs.get('Latitude_of_southern_pole'))
-            grid_info['o_lat_p'] = - center_lat[0] if center_lat[0] < 0 else 90 - center_lat[0]
-
-            grid_info['o_lon_p'] = 180
-
-        for attr in attrs:
-            bm_arg = ncl_to_basemap[attr]
-            val = lat.attrs[attr]
-            val = val[0] if isinstance(val, np.ndarray) else val
-            grid_info[bm_arg] = val
-            del val
-
-            if self.model == 'hrrrhi':
-                grid_info['lat_0'] = 20.44
-                grid_info['lon_0'] = 202.54
-                grid_info['width'] = 2000000
-                grid_info['height'] = 2000000
-
-        del lat
+            grid_info["lat_0"] = 20.44
+            grid_info["lon_0"] = 202.54
+            grid_info["width"] = 2000000
+            grid_info["height"] = 2000000
 
         return grid_info
 
-    def icing_adjust_trace(self, values, **kwargs):
-
-        # pylint: disable=unused-argument,no-self-use
-
-        ''' Changes the value of ICSEV trace from 4.0 to 0.5, to maintain ascending order '''
-
-        vals = np.where(values == 4.0, 0.5, values)
-
-        return vals
-
-    def run_max(self, values, **kwargs):
-
-        ''' Finds the max hourly value over all the forecast lead times available. '''
-
-        # pylint: disable=unused-argument,no-self-use
-
-        return values.max(dim='fcst_hr')
-
-    def run_min(self, values, **kwargs):
-
-        ''' Finds the min hourly value over all the forecast lead times available. '''
-
-        # pylint: disable=unused-argument,no-self-use
+    @staticmethod
+    def icing_adjust_trace(values: DataArray, **kwargs):  # noqa: ARG004
+        """Changes the value of ICSEV trace from 4.0 to 0.5, to maintain ascending order."""
 
-        return values.min(dim='fcst_hr')
+        return where(values == 4.0, 0.5, values)
 
-    def run_total(self, values, **kwargs):
+    @staticmethod
+    def run_max(values: DataArray, **kwargs):  # noqa: ARG004
+        """Finds the max hourly value over all the forecast lead times available."""
 
-        ''' Sums over all the forecast lead times available. '''
+        return values.max(dim="time")  # pragma: no cover
```

Comment:
Not testing this one since I don't have data available for multiple times in the tests dataset at this time. This is repeated for the next few time stats.